src/containers/user-profile-modal.tsx

```diff
@@ -10,32 +10,29 @@ import config from '@/config';
 import { useParticipant } from '@/providers/participant';
 import { type UserProfileInput, UserProfileSchema } from '@/types/schema.types';
 import { zodResolver } from '@hookform/resolvers/zod';
-import { useEffect, useState } from 'react';
 import { useForm } from 'react-hook-form';
 
-export default function UserProfileModal() {
+interface UserProfileModalProps {
+  isOpen?: boolean;
+  setIsOpen?: (isOpen: boolean) => void;
+}
+
+export default function UserProfileModal({ isOpen, setIsOpen }: UserProfileModalProps = {}) {
```

Comment:
If we were to assign the default value of the props. Please do this

```suggestion
export default function UserProfileModal({ isOpen = false, setIsOpen }: UserProfileModalProps) {
```

But if you do not have any intention to assign default values to props, you can do it like this.

```suggestion
export default function UserProfileModal({ isOpen, setIsOpen }: UserProfileModalProps) {
```